frontend/sw360-portlet/src/main/java/com/siemens/sw360/portal/common/AttachmentPortletUtils.java

```diff
@@ -264,4 +270,23 @@ private boolean alreadyHavePart(AttachmentContent attachment, ResumableUpload re
         }
     }
 
+    public Attachment getAttachmentForDisplay(User user, String attachmentContentId) {
+        try {
+            String filename = client.getAttachmentContent(attachmentContentId).getFilename();
+            return CommonUtils.getNewAttachment(user, attachmentContentId, filename);
+        } catch (TException e) {
+            log.error("Could not get attachment content", e);
+        }
+        return null;
+    }
+
+    public void deleteUnneededAttachments(Set<String> attachmentContentIds){
```

Comment:
this method would just as happily delete any needed attachments, not only unneeded. I'd call it `deleteAttachments`